WPMedia/fusion-news-theme-blocks
# Tmedia 113: atttribution hyphen fix

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
This Pull Request modifies the blockquote and pullquote blocks. Citations for these types of quote blocks were being rendered with empty content(author), so the code if modified to:

1. not render a hyphen when there is no citation content
2. modify the css to keep styles clean and even across blocks
3. include tests

## Jira Ticket
- [TMEDIA-113](https://arcpublishing.atlassian.net/browse/TMEDIA-113)

## Acceptance Criteria
Actual Behavior
If users include a block or pull quote within Composer and they don't use the attribution field, on the front end a hyphen still appears within the quote as if there were an attribution.

Expected Behavior
If there is no attribution associated with the quote, there should be no hyphen. Ideally, this piece of the quote element would collapse.

## Test Steps
1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l {blocks to link}`
3. Run tests and check visually
4. Go to https://corecomponents.arcpublishing.com/pf/local/2021/04/26/pull-quote-block-quote-test-story/?_website=the-gazette to see original without corrections
5. Go to http://localhost/pf/local/2021/04/26/pull-quote-block-quote-test-story/?_website=the-gazette to see local branch changes

## Effect Of Changes
### Before
Hyphen rendered with no author attached
Pullquote:
![Screen Shot 2021-04-27 at 4 01 46 PM](https://user-images.githubusercontent.com/82830431/116312342-edeff080-a771-11eb-9dc8-58c2ec93a979.png)
Blockquote:
![Screen Shot 2021-04-27 at 4 02 27 PM](https://user-images.githubusercontent.com/82830431/116312393-ff38fd00-a771-11eb-8038-dc9a16e6cd5b.png)


### After
Hyphen not rendered with empty citation content(no author)
Pullquote:
![Screen Shot 2021-04-27 at 3 46 40 PM](https://user-images.githubusercontent.com/82830431/116312059-93569480-a771-11eb-850a-73786225e7c1.png)
Blockquote:
![Screen Shot 2021-04-27 at 3 46 57 PM](https://user-images.githubusercontent.com/82830431/116312131-a9fceb80-a771-11eb-8f6c-96ca470e247f.png)


## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
